README.md

```diff
@@ -225,6 +225,7 @@ Each subcommand generally accepts the same options and syntax, and they mirror t
 * `--templatePath PATH` or `-p PATH`: define the Stencil template to use, using a full path.
 * Note: you should specify one and only one template when invoking SwiftGen. You have to use either `-t` or `-p` but should not use both at the same time (it wouldn't make sense anyway and you'll get an error if you try)
 * `--filter REGEX` or `-f REGEX`: the filter to apply to each input path. Each command has a default filter that you can override with this option.
+* Note: use `.*` to match any characters, and don't forget to escape the dot (`\.`) if you want to match a literal dot like for an extension. For example, use `.*\.xib` to match files with a `.xib` extension.
```

Comment:
I just realised: shouldn't we end all of our `defaultFilter` with `$`? (and maybe start it with `^`… or even simpler, don't even include `.*`?

Because our RegExes are not anchored to the whole file name when we match them, right? So it will match if part of the path matches, even if it's in the middle…

For example the `.*\.strings` regex actually matches `Loc.stringsdict` because `.strings` exists right in the middle.
While what we probably want is a RegEx like `.strings$` instead… so maybe we should rewrite all the `defaultFilters`… and also suggest that instead in that `Note` too…

Reply:
Also, didn't re-check but, does the `RegEx` match the whole path when filtering, on only the lastPathComponent (e.g. `basename`)? It might be important to decide what we prefer (filtering the basename is more simple and might avoid surprises, might also be less powerful, but regex on the whole path might not be worth it anyway) and specify which we chose in the docs.

Reply:
It matches the whole (relative) path. I'd say keep it, so that users can filter out specific folders if they really want.

You're right that I should add the end token, to ignore `stringsdict` (for now). I'd keep the `.*` to ensure we're matching a filename with that extension (and not something with no name, which could lead to other weird cases).

Reply:
Ok but if we match the whole (relative) path and not just the `basename`, that's indeed more powerful, but means that `^` will mean a different thing… and then how do you match "all files beginning with `SG`" for example? You can't use a RegEx like `^SG` because a file with relative path `foo/bar/SGBaz.json` won't match when you match the whole path. You can't use `/SG` either, because files at the root of the scanned directory will be `SGQux.json` and won't have the `/` then…

Reply:
`SG[^\/]*\.json` or something like that?

Reply:
Nope, that will match "paths containing `SG` followed by anything but a slash, followed by anything, following by `.json`"… so a totally different thing. For example your pattern will match `foo/bar/bazSGqux.json`…

Reply:
`^(.+\/|)SG[^\/]*\.json$` then? I dunno, it's late 😆

Reply:
Keep in mind that most people probably won't need more than the default provided filters, and if they ever need a custom filter, will probably have enough with some simple regexes with for example fixed folder names. For those one or 2 users that need something really specific, they can have some fun with regexes and ask a question here 🤷‍♂️ 

Let's also not forget that we allow multiple input paths, so that'll also cover a bunch of edge cases.